### PR TITLE
Power Armor Rework

### PR DESCRIPTION
**Pull request by BurpleBineapple**
## About The Pull Request

A bigger change to power armor's mechanics and projectile code. 

**TL;DR version: Projectiles with lower than 0.15 armor penetration now have a high chance to be completely blocked by power armor. Power armor's temperature values have also been fixed so that you don't die the second a cold/heat wave happens.**

Longer version of the whole thing, this PR has powered armors check for a projectile's armor penetration, and apply a block if it is lower than 0.15. Power armors have been given variables for a "block chance" and a "reflect chance".

How it works is that once you're hit by a projectile with the variable, the power armor runs a proc to check if the bullet is blocked or not. Salvaged armors have low values of about 20%. Functioning armors have values of 80% or higher. If the projectile is successfully blocked, it runs a parry and does one of two things. Either A: it deflects the projectile in a random direction based on the "deflect chance" or B: It stops the bullet and you take no damage.

The calibers this works on are:
- All 9mm except AP
- All 10mm except AP
- all 45ACP except AP
- all .22
- Regular shotgun buckshot
- Beanbag/Rubbershot shotgun rounds
- All hollow points (JHP)

357, 44, and 45-70 will work as normal. All rifles will work as normal. All lasers will work as normal. This new system means that power armor can no longer be reliably damaged by a lone wastelander with a 9mm Browning or a Grease Gun. AP bullets however will be **very** effective against power armor, encouraging their use if you plan to fight the Brotherhood of Steel.

Salvaged power armors also have these bonuses but to a much lesser extent, making the drastic speed debuff actually worth it now.

Later down the line I do plan to add a battery system for power armor as well, though that's a much harder task.

## Why It's Good For The Game

These values are taken from the Fallout 2 damage threshold and damage resistance values, which is how I selected which calibers should and shouldn't be blocked. The lowest power round that is guaranteed to go through power armor are 5.56 rifles and 357 handguns. Power armor is by no means invincible, but using low caliber bullets will not be very effective against the **pinnacle of pre-war military technology**. Rifles and AP rounds will still kill a Paladin in T-45d in less than a magazine, as normal.

Power armor will now truly be sought after, I would hope. Not just for the flex but for its superiority to normal armor. Even salvage will be worth the huge speed debuff.

## Changelog
:cl:
add: New variables to projectile code to set true or false to if power armor is able to block the bullet
add: New system for power armors to block low powered bullets based on an assigned percent chance
balance: Power armor is now stronger against pistol calibers, and encourages use of AP ammunition if fighting the Brotherhood of Steel
fix: Fixed power armor causing severe burns due to temperature changes
/:cl:

### CODE REVIEW

File: code/modules/clothing/suits/f13armor.dm
Comment (BurpleBineapple) [inline code/modules/clothing/suits/f13armor.dm:467]:
```suggestion
```

**Review by BurpleBineapple — COMMENTED** [review]
Somehow left the old armor value in, will delete

File: code/modules/projectiles/projectile.dm
Comment (MarinaGryphon) [inline code/modules/projectiles/projectile.dm:131]: This is just a bad idea. You should just use the `armour_penetration` var available on all subtypes of `/obj/item`. It ranges from 0 to 1; find a sufficient threshold to block bullets under. You could even set the threshold on each power armour subtype to give it varying amounts of armour toughness; some block anything under 80%, while some only block anything under 20%.

As it stands, adding an entirely new value to bullets is just silly.

File: code/modules/projectiles/projectile.dm
Comment (MarinaGryphon) [inline code/modules/projectiles/projectile.dm:844]: See above. This is, frankly, a silly proc as well; it's hyper-specific and all it does is run an istype and var access. If you still want to keep `power_armor_blocks` (you shouldn't), this should be either:
1) A proc on power armour, like `can_penetrate(atom/A)`, called in `run_block`. This is the ideal option.
or
2) Inlined in `run_block`, since it's just an istype and variable access:
```
	var/blocked = FALSE
	if(istype(A, /obj/item/projectile))
		var/obj/item/projectile/P = A
		blocked = P.power_armor_blocks
	if(blocked && ...)
```
etc.

File: code/modules/clothing/head/f13head.dm
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:236]: If you want to do the sane thing and change it to checking armour penetration on things it's hit with (as the var is available on anything derived from `/obj/item`), this should probably be replaced with
```suggestion
	var/armour_block_threshold = 0.75
```
or something similar.
Note that you should probably use a spreadsheet of AP values for this in order to fine-tune what guns hit what armours. It should feel much more natural if any extra ammo types are added, *and* it should also handle things that aren't subtypes of projectiles, i.e. a scrap launcher or something.

File: code/modules/clothing/head/f13head.dm
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:237]:
```suggestion
	var/const/list/protected_zones = list(BODY_ZONE_HEAD)
```
Not redefined on any subtypes, not permuted at runtime; this should be a const var.

File: code/modules/clothing/head/f13head.dm
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:238]: I would probably rework this to be based on armour penetration; if something hits with high enough force and low enough armour penetration, that force should be redirected into a ricochet.

~~Alternatively, remove it entirely. Fallout 2 doesn't have deflection.~~

File: code/modules/clothing/suits/f13armor.dm
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:212]:
```suggestion
	var/const/list/protected_zones = list(BODY_ZONE_CHEST, BODY_ZONE_PRECISE_GROIN, BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
```
Not redefined on any subtypes, not permuted at runtime; this should be a const var.

File: code/modules/clothing/suits/f13armor.dm
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:261]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:286]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:310]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:334]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:357]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:380]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:403]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:448]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:471]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:522]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:561]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:329]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:352]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:374]: Duplicated code; use inheritance.

**Comment by BurpleBineapple** [comment]
Need to fix a few things with messy code that someone pointed out. I actually may have had a stroke

File: code/modules/clothing/head/f13head.dm
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:397]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:439]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:462]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:511]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:537]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:559]: Duplicated code; use inheritance.

**Review by MarinaGryphon — CHANGES_REQUESTED** [review]

**Comment by BurpleBineapple** [comment]
Should be ready for review once more

**Comment by BurpleBineapple** [comment]
stiiiill some minor bugs with this one, so don't merge

**Comment by BurpleBineapple** [comment]
After further testing, it should be fine to merge. I was checking on an outdated branch.

**Pull request merged**
